api-server/src/server/boot/challenge.js

```diff
@@ -82,6 +90,340 @@ export default async function bootChallenge(app, done) {
+  );
+  return Observable.fromPromise(updatePromise)
+    .map(() => {
+      return res.json({
+        submittedChallenges,
+        points
+      });
+    })
+    .subscribe(() => {}, next);
+}
+
+/**
+ * Handles submissions for all certification projects.
+ * @param {Request} req
+ * @param {Response} res
+ * @returns {Observable}
+ */
+export function projectCompleted(req, res, next) {
+  const { user, body } = req;
+  const { files, completedDate = Date.now(), id } = body;
+  let completedChallenge = body;
+
+  let finalChallenge;
+  const $push = {},
+    $set = {},
+    $pull = {};
+  const {
+    completedChallenges = [],
+    needsModeration = false,
+    savedChallenges = []
+  } = user;
+
+  const oldIndex = completedChallenges.findIndex(
+    ({ id: challengeId }) => challengeId === id
+  );
+
+  const alreadyCompleted = oldIndex !== -1;
+  const oldChallenge = alreadyCompleted ? completedChallenges[oldIndex] : null;
+
+  if (alreadyCompleted) {
+    finalChallenge = {
+      ...completedChallenge,
+      completedDate: oldChallenge.completedDate
+    };
+    $set[`completedChallenges.${oldIndex}`] = finalChallenge;
+  } else {
+    finalChallenge = {
+      ...completedChallenge
+    };
+    $push.progressTimestamps = completedDate;
+    $push.completedChallenges = finalChallenge;
+  }
+
+  if (savableChallenges.includes(id)) {
+    const challengeToSave = {
+      id,
+      lastSavedDate: completedDate,
+      files: files?.map(file =>
+        pick(file, ['contents', 'key', 'name', 'ext', 'history'])
+      )
+    };
+
+    const savedIndex = savedChallenges.findIndex(
+      ({ id: challengeId }) => challengeId === id
+    );
+
+    if (savedIndex >= 0) {
+      $set[`savedChallenges.${savedIndex}`] = challengeToSave;
+      savedChallenges[savedIndex] = challengeToSave;
+    } else {
+      $push.savedChallenges = challengeToSave;
+      savedChallenges.push(challengeToSave);
+    }
+  }
+
+  // remove from partiallyCompleted on submit
+  $pull.partiallyCompletedChallenges = { id };
+
+  if (needsModeration) $set.needsModeration = true;
+
+  const updateData = {};
+  if (!isEmpty($set)) updateData.$set = $set;
+  if (!isEmpty($push)) updateData.$push = $push;
+  if (!isEmpty($pull)) updateData.$pull = $pull;
+
+  // ----------------------------------------------
+
+  const points = alreadyCompleted ? user.points : user.points + 1;
+  const updatePromise = new Promise((resolve, reject) =>
+    user.updateAttributes(updateData, (err, instance) => {
+      if (err) {
+        return reject(err);
+      }
+      return resolve(instance);
+    })
+  );
+  return Observable.fromPromise(updatePromise)
+    .map(() => {
+      return res.json({
+        completedChallenges: updateData.$push.completedChallenges,
+        points,
+        savedChallenges
+      });
+    })
+    .subscribe(() => {}, next);
+}
+
+export const expectedProjectStructures = {
+  [challengeTypes.multifileCertProject]: [
+    // 14 'responsive-web-design'
```

Comment:
These comments are a little confusing and will definitely become obsolete (this won't just be RWD for long).  How about putting them as examples in `challenge-types.js`?